handlers/feedback.go

```diff
@@ -140,69 +95,112 @@ func addFeedback(w http.ResponseWriter, req *http.Request, isPositive bool, rend
 	decoder := schema.NewDecoder()
 	decoder.IgnoreUnknownKeys(true)
 
-	var f Feedback
-	if err := decoder.Decode(&f, req.Form); err != nil {
+	var ff model.FeedbackForm
+	if err := decoder.Decode(&ff, req.Form); err != nil {
 		log.Error(ctx, "unable to decode request form", err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	if f.Description == "" && !isPositive {
-		getFeedback(w, req, f.URL, "description", f.Description, f.Name, f.Email, lang, rend, cacheService)
+	validationErrors := validateForm(&ff)
+	if len(validationErrors) > 0 {
+		getFeedback(w, req, validationErrors, ff, lang, rend, cacheService)
 		return
 	}
 
-	if len(f.Email) > 0 && !isPositive {
-		if ok, err := regexp.MatchString(`^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,6}$`, f.Email); !ok || err != nil {
-			getFeedback(w, req, f.URL, "email", f.Description, f.Name, f.Email, lang, rend, cacheService)
-			return
-		}
-	}
-
-	if f.URL == "" {
-		f.URL = "Whole site"
+	if ff.URL == "" {
+		ff.URL = "Whole site"
 	}
 
 	if err := emailSender.Send(
 		from,
 		[]string{to},
-		generateFeedbackMessage(f, from, to, isPositive),
+		generateFeedbackMessage(ff, from, to),
 	); err != nil {
 		log.Error(ctx, "failed to send message", err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	returnTo := f.URL
+	returnTo := ff.URL
 
 	if returnTo == "Whole site" || returnTo == "" {
 		returnTo = "https://www.ons.gov.uk"
 	}
 
-	redirectURL := "/feedback/thanks?returnTo=" + returnTo
+	redirectURL := fmt.Sprintf("/feedback/thanks?returnTo=%s", returnTo)
 	http.Redirect(w, req, redirectURL, http.StatusMovedPermanently)
 }
 
-func generateFeedbackMessage(f Feedback, from, to string, isPositive bool) []byte {
-	var description string
-	if isPositive {
-		description = "Positive feedback received"
-	} else {
-		description = f.Description
+// validateForm is a helper function that validates a slice of FeedbackForm to determine if there are form validation errors
+func validateForm(ff *model.FeedbackForm) (validationErrors []core.ErrorItem) {
+	if ff.Type == "" && ff.FormLocation != "footer" {
+		validationErrors = append(validationErrors, core.ErrorItem{
+			Description: core.Localisation{
+				LocaleKey: "FeedbackChooseType",
+				Plural:    1,
+			},
+			URL: "#radio-error",
```

Comment:
Might be nice to have this as a less generic ID (like #type-error) in case we add other inputs later - interested on your view.

Reply:
Can't hurt :)